frontend/model/contracts/group.js

```diff
@@ -132,31 +151,25 @@ DefineContract({
       }
     },
     'gi.contracts/group/proposal': {
-      validate: data => {
-        objectOf({
-          proposalType: proposalType,
-          proposalData: object, // data for Vue widgets
-          votingRule: ruleType,
-          expires_date_ms: number // calculate by grabbing proposal expiry from group properties and add to `meta.createdDate`
-        })
+      validate: objectOf({
+        proposalType: proposalType,
+        proposalData: object, // data for Vue widgets
+        votingRule: ruleType,
+        expires_date_ms: number // calculate by grabbing proposal expiry from group properties and add to `meta.createdDate`
+      }),
+      process (state, { data, meta, hash }) {
         // make sure this isn't a duplicate proposal
-        const ourUsername = sbp('state/vuex/getters').ourUsername
-        const groupID = sbp('state/vuex/state').currentGroupId
-        const groupState = sbp('state/vuex/state')[groupID]
-        for (const hash in groupState.proposals) {
-          const prop = groupState.proposals[hash]
+        for (const hash in state.proposals) {
+          const prop = state.proposals[hash]
           if (prop.status === STATUS_OPEN &&
             prop.data.proposalType === data.proposalType &&
-            prop.meta.username === ourUsername &&
+            prop.meta.username === meta.username &&
```

Comment:
Why this validation is not done during process instead of validate?

Reply:
Because we cannot access state that exists outside of the contract (submodule) state, and `sbp('state/vuex/getters').ourUsername` does that